mlflow/data/huggingface_dataset.py

```diff
@@ -0,0 +1,233 @@
+
+    @property
+    def ds(self) -> datasets.Dataset:
+        """
+        The Hugging Face `datasets.Dataset` instance.
+
+        :return: The Hugging Face `datasets.Dataset` instance.
+        """
+        return self._ds
+
+    @property
+    def targets(self) -> Optional[str]:
+        """
+        The name of the Hugging Face dataset column containing targets.
+
+        :return: The Hugging Face `datasets.Dataset` instance.
+        """
+        return self._targets
+
+    @property
+    def source(self) -> HuggingFaceDatasetSource:
+        """
+        Hugging Face dataset source information.
+
+        :return: A HuggingFaceDatasetSource instance.
+        """
+        return self._source
+
+    @property
+    def profile(self) -> Optional[Any]:
+        """
+        Summary statistics for the Hugging Face dataset, including the number of rows,
+        size, and size in bytes.
+        """
+        return {
+            "num_rows": self._ds.num_rows,
+            "dataset_size": self._ds.dataset_size,
+            "size_in_bytes": self._ds.size_in_bytes,
+        }
+
+    @cached_property
+    def schema(self) -> Schema:
+        """
+        The MLflow ColSpec schema of the Hugging Face dataset.
+        """
+        try:
+            df = next(
+                self._ds.to_pandas(
+                    batch_size=_MAX_ROWS_FOR_DIGEST_COMPUTATION_AND_SCHEMA_INFERENCE, batched=True
+                )
+            )
+            return _infer_schema(df)
+        except Exception as e:
+            _logger._warning("Failed to infer schema for Hugging Face dataset. Exception: %s", e)
+            return None
+
+    def to_pyfunc(self) -> PyFuncInputsOutputs:
+        df = self._ds.to_pandas()
+        if self._targets is not None:
+            if self._targets not in df.columns:
+                raise MlflowException(
+                    f"Failed to convert Hugging Face dataset to pyfunc inputs and outputs because"
+                    f" the pandas representation of the Hugging Face dataset does not contain the"
+                    f" specified targets column '{self._targets}'.",
+                    # This is an internal error because we should have validated the presence of
+                    # the target column in the Hugging Face dataset at construction time
+                    INTERNAL_ERROR,
+                )
+            inputs = df.drop(columns=self._targets)
+            outputs = df[self._targets]
+            return PyFuncInputsOutputs(inputs=inputs, outputs=outputs)
+        else:
+            return PyFuncInputsOutputs(inputs=df, outputs=None)
+
+
+def from_huggingface(
+    ds: datasets.Dataset,
+    path: str,
+    targets: Optional[str] = None,
+    data_dir: Optional[str] = None,
+    data_files: Optional[Union[str, Sequence[str], Mapping[str, Union[str, Sequence[str]]]]] = None,
+    revision: Optional[Union[str, datasets.Version]] = None,
+    task: Optional[Union[str, datasets.TaskTemplate]] = None,
```

Comment:
Unfortunately, this information cannot be inferred from the HuggingFace dataset. It would be nice if it could. For now, asking the user to at least specify `path` seems okay enough. I toyed with the idea of having the user specify a `datasets.DatasetBuilder` class, but this seems like more trouble than it's worth for users (having to construct another class just to use this API).